### DISCUSSION TITLE
Does KafkaConnect use GCP's Workload Identity Federation?

### DISCUSSION THREAD

**Discussion by jymaeng95** [strimzi/strimzi-kafka-operator#9414] [Q&A] [answered]
I want to use GCP Pub/Sub Source Connector using KafkaConnect, KafkaConnector CR.

GCP recommends using the **workload identity federation** to access resources in the cloud.

At this time, ServiceAccountToken should be mounted on the projected volume at the time the pod was created, but Strimzi KafkaConnect Spec could not determine the specification for mounting the projected volume and the specification for defining ServiceAccountName.

**Is there any way I can use workload identity federation on KafkaConnect?**

This is a link to the GCP guide I referenced.
> https://cloud.google.com/iam/docs/workload-identity-federation-with-kubernetes#eks

Thanks.

**Comment by scholzj** [comment, marked as answer]
I think you might need to translate the Google doc to the Kubernetes level and share it here for people who have no experience with _GCP's Workload Identity Federation_ - to explain what exactly is it you need / want. In general, Strimzi allows you to mount Config Maps or Secrets as volumes or environment variables. You can also - as with any Kafka Connect - use configuration providers.
